### DISCUSSION TITLE
Docker container crash and restart all time

### DISCUSSION THREAD

**Discussion by MitsuhaKitsune** [open-webui/open-webui#1018] [General]
# Bug Report
When I try to start the docker container following the readme and guide, the container start to restart in loop constantly.

## Description

**Bug Summary:**
Docker container crash and restart on startup

**Steps to Reproduce:**
Just run this command:

`docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_API_BASE_URL=http://192.168.1.10:11434/api --name open-webui --restart always ghcr.io/open-webui/open-webui:main`

**Expected Behavior:**
Docker container start successfull and let me open the web UI

**Actual Behavior:**
Docker container crash and restart on startup

## Environment

- **Operating System:** Debian 12
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

## Reproduction Details

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I have reviewed the troubleshooting.md document.
- [x] I have included the browser console logs.
- [X] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
```
 No WEBUI_SECRET_KEY provided
 

 Generating WEBUI_SECRET_KEY
 

 Loading WEBUI_SECRET_KEY from .webui_secret_key
 

 /usr/local/lib/python3.11/site-packages/pydantic/_internal/_fields.py:151: UserWarning: Field "model_max_budget" has conflict with protected namespace "model_".
 

 
 

 You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
 

   warnings.warn(
 

 No WEBUI_SECRET_KEY provided
 

 Generating WEBUI_SECRET_KEY
 

 Loading WEBUI_SECRET_KEY from .webui_secret_key
 

 /usr/local/lib/python3.11/site-packages/pydantic/_internal/_fields.py:151: UserWarning: Field "model_max_budget" has conflict with protected namespace "model_".
 

 
 

 You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
 

   warnings.warn(
 

 No WEBUI_SECRET_KEY provided
 

 Loading WEBUI_SECRET_KEY from .webui_secret_key
 

 /usr/local/lib/python3.11/site-packages/pydantic/_internal/_fields.py:151: UserWarning: Field "model_max_budget" has conflict with protected namespace "model_".
 

 
 

 You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
 

   warnings.warn(
```
 

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Docker

## Additional Information

Logs don't provide any aditional info

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by MitsuhaKitsune** [comment]
I try to run the webui from source code, and I can see "Illegal instruction" that crash the program.

Uncompatible hardware maybe? I explain better my enviroment:

I just trying to run the webui on one of my servers of my LAN (HP ProLiant MicroServer G7 HSTNS 5151 N54L), but didn't have a recent CPU or GPU to run AI, just want the UI, the ollama server are on other hardware of my LAN.

**Reply by warm3snow** [reply]
me too
```
INFO  [open_webui.env] Embedding model set: sentence-transformers/all-MiniLM-L6-v2
WARNI [langchain_community.utils.user_agent] USER_AGENT environment variable not set, consider setting it to identify your requests.

  ___                    __        __   _     _   _ ___
 / _ \ _ __   ___ _ __   \ \      / /__| |__ | | | |_ _|
| | | | '_ \ / _ \ '_ \   \ \ /\ / / _ \ '_ \| | | || |
| |_| | |_) |  __/ | | |   \ V  V /  __/ |_) | |_| || |
 \___/| .__/ \___|_| |_|    \_/\_/ \___|_.__/ \___/|___|
      |_|


v0.5.4 - building the best open-source AI user interface.

https://github.com/open-webui/open-webui

[1]    36504 illegal hardware instruction  open-webui serve

```

**Comment by tjbck** [comment]
Hmm, unable to reproduce the issue at all. Might have something to do with your hardware configuration, moving to discussion.

**Comment by MitsuhaKitsune** [comment]
Illegal Instruction are produced when a processor didn't support the instruction set of software, the CPU of this server its a [AMD Turion II Neo N54L](https://www.cpu-world.com/CPUs/K10/AMD-Turion%20II%20Neo%20N54L%20-%20TEN54LSDV23GME.html).

As I know, llama as like the most AI projects need atleast AVX2 CPU's or cuda GPU, but I only running this project on the server (the ollama server are in another PC).

I try to connect from this server to the ollama in a simple vue webpage through the API and connection works, so I discard any connection problem.

As I understand, this project don't have any AI process, its just a web interface to connect to ollama API, so seems strange that throw this error, but some dependency can produce incompatibilities with this CPU if need more recent instruction sets.

Unafortunally I try to do a fast research but can't found what produce this.

**Reply by tjbck** [reply]
Try installing without docker: https://docs.openwebui.com/getting-started/#how-to-install-without-docker

**Reply by MitsuhaKitsune** [reply]
I tried it, but unafortunally crash on startup too, the unic difference between docker and source code, it's that on source code I can read "Illegal Instruction" before crash,

**Comment by amartinAIDIMME** [comment]
I have a similar problem in windows:
https://github.com/open-webui/open-webui/discussions/942

**Reply by bladethirst** [reply]
I hava a similar problem in centos 7.9. I run it with docker.

**Reply by bladethirst** [reply]
I found the problem related to docker version.  with parameter "--security-opt seccomp:unconfined" or "–privileged=true" when running container can fix the problem.

**Comment by elabz** [comment]
@bladethirst , could you please specify what exactly you did to fix the problem? My attempts to run with `--security-opt seccomp:unconfined` did not change anything at all - container keeps restarting. 
```
docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_API_BASE_URL=http://192.168.0.151:11434/api --name open-webui-1 --restart always --security-opt seccomp:unconfined ghcr.io/open-webui/open-webui:main
```
adding `privileged: true` to `docker-compose.yaml` also yielded no results

Would appreciate your input on this. Thanks!

**Reply by justinh-rahb** [reply]
Considering @bladethirst is using CentOS, it's likely his problem was caused by AppArmor preventing an unprivileged container from accessing the `host` network. Are you also using CentOS or another distro with AppArmor enabled?

I think both of you would be better off to investigate why the normal overlay network and published port method isn't working, as setting `--network=host` is often a band-aid fix for something else.

**Reply by elabz** [reply]
@justinh-rahb  - I am not sure network is the problem as the docker-compose.yaml it starts off of (if not using the command above) has
```
    extra_hosts:
      - host.docker.internal:host-gateway
```
And leads to the same constant restarting

**Reply by elabz** [reply]
Oh, and yes, I am on Ubuntu 22.04 server 
`Linux ash 5.15.0-97-generic #107-Ubuntu SMP Wed Feb 7 13:26:48 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
`

**Comment by fevolq** [comment]
我也遇到了这个问题，在基于wsl的ubuntu机器上，通过docker启动的，网页无法访问

**Comment by elabz** [comment]
@tjbck , could we move this back to issues please? It's still happening on the latest update, with or without GPU. All the latest CUDA and NVIDIA drivers (although this should not be affecting anything on `open-webui` container?

Can someone suggest a way to enable more descriptive error log? The `docker logs open-webui` only contains warnings, not errors (mentioned above in this thread)

**Comment by elabz** [comment]
There is an error on dmesg that may correlate to this - uvicorn breaks on something:

```
traps: uvicorn[20563] trap invalid opcode ip:7f07bf3597cc sp:7ffcf5a50f88 error:0 in libctranslate2-de03ae65.so.4.0.0[7f07bf206000+2d7e000]
``` 
Could this be relevant?

**Reply by tjbck** [reply]
Same issue when running it locally without docker?

**Reply by elabz** [reply]
It is. 

Could this be a low RAM situation? I am installing on 8GB RAM machine

**Reply by tjbck** [reply]
If you're running both Ollama and Open WebUI on the same machine, I highly suspect that low RAM size is the cause.

**Reply by elabz** [reply]
🤔 I turned out to be 6GB, not even 8GB. However, shutting down ollama and only starting open-webui does not fix anything. I ran it with `top` running and the memory usage never goes above 1GB

**Reply by elabz** [reply]
... and another similar question: can it run on CPU with 4 cores? 

```
$ lshw -c cpu
WARNING: you should run this program as super-user.
  *-cpu
       product: AMD Phenom(tm) 9550 Quad-Core Processor
       vendor: Advanced Micro Devices [AMD]
       physical id: 1
       bus info: cpu@0
       version: 16.2.3
       width: 64 bits
       capabilities: fpu fpu_exception wp vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp x86-64 3dnowext 3dnow constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid pni monitor cx16 popcnt lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs hw_pstate vmmcall npt lbrv svm_lock
       configuration: microcode=16777357
WARNING: output may be incomplete or inaccurate, you should run this program as super-user.

```

**Reply by justinh-rahb** [reply]
That CPU is old enough to get a driver's license! 👀 Sorry but time to upgrade, your CPU doesn't even support AVX1, let alone AVX2.

**Reply by elabz** [reply]
Thank you for checking, @justinh-rahb , I am still very curious about running ollama on it, perhaps serving open-webui from a different machine

**Reply by justinh-rahb** [reply]
Welcome to give that a go, but I don't see it going much better. Without at least some acceleration like AVX, llama.cpp is dog slow.

**Comment by kelley01** [comment]
I am having the same issue. I had WebUI working, then I changed my Docker root directory from /var/lib/docker to a secondary hard drive. Now open-webui will not stay up. It keeps stopping and restarting. Other Docker containers are working just fine.

**Comment by viba1** [comment]
Same problem here from a fresh docker installation. I noticed that the problem systematically occurs when I download files.

**Comment by AnzacExodus** [comment]
Same thing was happening to me running open webui through docker. What worked was running the following commands: 

docker stop open-webui
docker rm open-webui
docker volume rm open-webui
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -e OLLAMA_BASE_URL=http://host.docker.internal:11434 -v open-webui:/app/backend/data --name open-webui --restart always --pull always ghcr.io/open-webui/open-webui:main

I'll link the original post where this solution was found:
https://github.com/ollama/ollama/issues/5892

**Comment by MohammadBnei** [comment]
I am having this issue today, more than one year later. I updated from 0.6.18 to 0.6.22, and suddenly its crashloopbackoff (i'm deploying on a kubernetes cluster). I cannot rollback, the issue is still happening. Very unfortunate

**Reply by MohammadBnei** [reply]
I changed the node it was running on to a node with recent hardware and it's working now. I guess we cannot have python and old computers mixing.

**Reply by duffyjp** [reply]
I have it installed in Podman on an M1 mac, and it's crash looping for me.  I don't see any error messages, it's just doing it's thing talking to my LLM server and then suddenly it's restarting.

That "Loading WEBUI_SECRET_KEY" part is the first line that runs if you start it normally.

The only other docker host I have is significantly older, though it's intel at least so more common.

```
�2025-08-25 22:35:50.945 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 192.168.127.1:50898 - "GET /api/v1/chats/?page=1 HTTP/1.1" 200
�2025-08-25 22:35:50.951 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 192.168.127.1:50898 - "GET /api/v1/folders/ HTTP/1.1" 200
�2025-08-25 22:36:18.981 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 192.168.127.1:37986 - "GET /_app/version.json HTTP/1.1" 200
�2025-08-25 22:36:30.825 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 192.168.127.1:31701 - "POST /api/chat/completions HTTP/1.1" 200
Loading WEBUI_SECRET_KEY from file, not provided as an environment variable.
Loading WEBUI_SECRET_KEY from .webui_secret_key
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [open_webui.env] 'ENABLE_SIGNUP' loaded from the latest database entry
WARNI [open_webui.env] 

WARNING: CORS_ALLOW_ORIGIN IS SET TO '*' - NOT RECOMMENDED FOR PRODUCTION DEPLOYMENTS.

INFO  [open_webui.env] VECTOR_DB: chroma
INFO  [open_webui.env] Embedding model set: sentence-transformers/all-MiniLM-L6-v2
�WARNI [langchain_community.utils.user_agent] USER_AGENT environment variable not set, consider setting it to identify your requests.

� ██████╗ ██████╗ ███████╗███╗   ██╗    ██╗    ██╗███████╗██████╗ ██╗   ██╗██╗
�██╔═══██╗██╔══██╗██╔════╝████╗  ██║    ██║    ██║██╔════╝██╔══██╗██║   ██║██║
�██║   ██║██████╔╝█████╗  ██╔██╗ ██║    ██║ █╗ ██║█████╗  ██████╔╝██║   ██║██║
�██║   ██║██╔═══╝ ██╔══╝  ██║╚██╗██║    ██║███╗██║██╔══╝  ██╔══██╗██║   ██║██║
�╚██████╔╝██║     ███████╗██║ ╚████║    ╚███╔███╔╝███████╗██████╔╝╚██████╔╝██║
� ╚═════╝ ╚═╝     ╚══════╝╚═╝  ╚═══╝     ╚══╝╚══╝ ╚══════╝╚═════╝  ╚═════╝ ╚═╝


v0.6.25 - building the best AI user interface.
```
